checkov/common/util/http_utils.py

```diff
@@ -134,3 +136,46 @@ def request_wrapper(
     else:
         raise Exception("Unexpected behavior: the method \'request_wrapper\' should be terminated inside the above for-"
                         "loop")
+
+
+async def aiohttp_client_session_wrapper(
+    url: str,
+    headers: dict[str, Any],
+    payload: dict[str, Any]
+) -> int:
+    request_max_tries = int(os.getenv('REQUEST_MAX_TRIES', 3))
+    sleep_between_request_tries = float(os.getenv('SLEEP_BETWEEN_REQUEST_TRIES', 1))
+
+    # adding retry mechanism for avoiding the next repeated unexpected issues:
+    # 1. Gateway Timeout from the server
+    # 2. ClientOSError
+    async with aiohttp.ClientSession(connector=aiohttp.TCPConnector(resolver=aiohttp.AsyncResolver())) as session:
+        for i in range(request_max_tries):
+            logging.info(f"[http_utils](aiohttp_client_session_wrapper) reporting attempt {i + 1} out of {request_max_tries}")
+            try:
+                async with session.post(
+                        url=url, headers=headers, json=payload
+                ) as response:
+                    content = await response.text()
+                if response.ok:
+                    logging.info(f"[http_utils](aiohttp_client_session_wrapper) - done successfully to url: \'{url}\'")
+                    return 0
+                elif i != request_max_tries - 1:
+                    await asyncio.sleep(sleep_between_request_tries * (i + 1))
+                    continue
+                else:
+                    logging.error(f"[http_utils](aiohttp_client_session_wrapper) - Failed to send report to "
+                                  f"url \'{url}\'")
+                    logging.error(f"Status code: {response.status}, Reason: {response.reason}, Content: {content}")
+                    return 1
+            except aiohttp.ClientOSError:
```

Comment:
Could there be other exceptions beside ClientOSError?
In that case, it could be beneficial to add a general except without a specific exception to catch.

Reply:
i think that we are better to control what exception are catched.and they are only those that we know that they may happen
let's say we got ZeroDivisionError somewhow, we will not to catch it, but raising it

Reply:
fixed